### PR TITLE
Validator Set Changes

### PR DESCRIPTION
**Pull request by ebuchman**
Very simple approach to validator set changes.

Still needs testing. Also will need more work for proper light client validation.

### CODE REVIEW

File: consensus/common_test.go [hunk 1]
```diff
@@ -266,6 +270,31 @@ func randConsensusNet(nValidators int) []*ConsensusState {
 	return css
 }
 
+// nPeers = nValidators + nNotValidator
+func randConsensusNetWithPeers(nValidators int, nPeers int) []*ConsensusState {
```
Comment (jaekwon) [inline consensus/common_test.go]: This isn't used?

**Comment by ebuchman** [comment]
@jaekwon please continue to review when you get a chance, but I'm finding it simpler to merge things to develop sooner

**Comment by jaekwon** [comment]
Was that necessary?  The branch is broken and so is ethermint.

I think develop should be semi-sacred.  If it's broken, it should be fixed asap (for other developers).

I imagine it's easy to overlook as well, since the pull request appears to be closed.  I only found this because I was investigating why ethermint was broken -- AppHashIsStale never gets set to true after commit.

**Comment by jaekwon** [comment]
It also appears to break Github's review system.  The previous review comments don't appear anymore when I go into the file-changed-diff tab.

**Comment by ebuchman** [comment]
Breaking githubs review system is reason enough to not do this. Didn't realize that would happen. 

AppHashIsStale gets set to true before Commit, when the state is updated but we don't have the new app hash (it becomes stale). What problem did you find?

**Comment by ebuchman** [comment]
I see you mean set to false. Thanks for the fix.

**Pull request merged**